src/lib.rs

```diff
@@ -393,5 +401,45 @@ pub mod pallet {
             bytes_to_hash.append(&mut bytes_in_name);
             blake2_256(&bytes_to_hash[..])
         }
+
+        fn validate_block_number(
+            valid_for: Option<T::BlockNumber>,
+        ) -> Result<T::BlockNumber, DidError> {
+            let max_block: T::BlockNumber = u32::max_value().into();
```

Comment:
Is that possible to use the generic way to check the max value? We might set up the T::BlockNumber as the u64 or other type in the runtime module; therefore, we shouldn't convert it into the u32.

T::BlockNumber already has the Bounded trait; I think we can use it directly.

```
        type BlockNumber: Parameter                                                                      
            + Member                                                                                     
            + MaybeSerializeDeserialize                                                                  
            + Debug                                                                                      
            + MaybeDisplay                                                                               
            + AtLeast32BitUnsigned                                                                       
            + Default                                                                                                             
            + Bounded                                                                                
            + Copy                                                                                   
            + sp_std::hash::Hash                                                                     
            + sp_std::str::FromStr                                                                   
            + MaybeMallocSizeOf                                                                      
            + MaxEncodedLen                                                                          
            + TypeInfo;                                                                              
```

Reply:
Yep, agreed... had this thought too.

Reply:
Any thoughts on this?

Reply:
Could we use `checked_add`?

Reply:
Good catch...updating it now. Thanks

Reply:
Updated. Please check.